aten/src/ATen/cpu/vec256/vec256_complex_double.h

```diff
@@ -0,0 +1,328 @@
+#pragma once
+
+#include <ATen/cpu/vec256/intrinsics.h>
+#include <ATen/cpu/vec256/vec256_base.h>
+#if defined(__AVX__) && !defined(_MSC_VER)
+#include <sleef.h>
+#endif
+
+namespace at {
+namespace vec256 {
+// See Note [Acceptable use of anonymous namespace in header]
+namespace {
+
+template <> class Vec256<std::complex<double>> {
+private:
+  std::complex<double> values[2] = {0};
+public:
+  using value_type = std::complex<double>;
+  static constexpr int size() {
+    return 2;
+  }
+  Vec256() {}
+  Vec256(std::complex<double> val) {
+    for (int i = 0; i != size(); i++) {
+      values[i] = val;
+    }
```

Comment:
These operations haven't been vectorized at all! The loops aren't even unrolled!

In other words, what it seems like what you've implemented here is not the equivalent of `vec256_float.h` (which is all vectorized) but `vec256_base.h` (which provides non-vectorized implementations of all operations to default back to.) So, I have two questions:

1. Can we rename the header so this is clear?
2. Can we deduplicate `vec256_complex_{double,float}`, since if these are non-vectorized, the implementations should basically be the same. This is similar to how `base` is only written once.